extension/src/setup/index.ts

```diff
@@ -604,22 +646,54 @@ export class Setup
     return setContextValue(ContextKey.STUDIO_CONNECTED, isConnected)
   }
 
-  private getSecret(key: string) {
-    const secrets = this.getSecrets()
-    return secrets.get(key)
-  }
+  private watchDvcConfigs() {
+    const createWatcher = (watchedPath: string) =>
+      createFileSystemWatcher(
+        disposable => this.dispose.track(disposable),
+        getRelativePattern(watchedPath, '**'),
+        path => {
+          if (
+            path.endsWith(join('dvc', 'config')) ||
+            path.endsWith(join('dvc', 'config.local'))
+          ) {
+            void this.updateIsStudioConnected()
+          }
+        }
+      )
 
-  private storeSecret(key: string, value: string) {
-    const secrets = this.getSecrets()
-    return secrets.store(key, value)
-  }
+    const globalConfigPath = getDVCAppDir()
+
+    createWatcher(globalConfigPath)
 
-  private removeSecret(key: string) {
-    const secrets = this.getSecrets()
-    return secrets.delete(key)
+    for (const workspaceFolder of getWorkspaceFolders()) {
+      createWatcher(workspaceFolder)
+    }
   }
 
-  private getSecrets() {
-    return this.secrets
+  private async setStudioAccessToken() {
+    if (!this.getCliCompatible()) {
+      this.studioAccessToken = undefined
+      return
+    }
+
+    if (this.dvcRoots.length !== 1) {
```

Comment:
[F] same thing here we are not supporting `--local` wrt to multiple projects being in the workspace.